### PR TITLE
Support Verifiable Credentials in Cred Manifest Flow

### PR DESCRIPTION
**Pull request by decentralgabe**
- add vcs to CM flows
- fixed a bug in validating credentials against schemas
- updated tests accordingly

### CODE REVIEW

File: pkg/server/router/manifest.go [hunk 1]
```diff
@@ -178,16 +179,22 @@ func (mr ManifestRouter) DeleteManifest(ctx context.Context, w http.ResponseWrit
 }
 
 type SubmitApplicationRequest struct {
-	Application manifestsdk.CredentialApplication `json:"application" validate:"required"`
 	// Once we have JWT signed wrapper that can get the did this can be removed
-	ApplicantDID string `json:"applicantDid" validate:"required"`
+	ApplicantDID string                            `json:"applicantDid" validate:"required"`
+	Application  manifestsdk.CredentialApplication `json:"credential_application" validate:"required"`
+	Credentials  []interface{}                     `json:"verifiableCredentials" validate:"required"`
```
Comment (nitro-neal) [inline pkg/server/router/manifest.go:185]: Awesome

File: internal/credential/verification.go [hunk 2]
```diff
@@ -5,86 +5,145 @@ import (
...
 		return util.LoggingErrorMsg(err, "could not create verifier")
 	}
+
+	// verify the signature on the credential
 	if err := verifier.Verify(keyaccess.JWKToken{Token: token}); err != nil {
 		return util.LoggingErrorMsg(err, "could not verify credential's signature")
 	}
-	if err = v.verifier.VerifyCredential(*cred); err != nil {
+
+	// if the credential has a schema, resolve it before it is to be used in verification
```
Comment (nitro-neal) [inline internal/credential/verification.go:81]: same code block on line 131, you could make into a single function

Comment (decentralgabe) [reply internal/credential/verification.go:81]: good idea

File: internal/credential/model.go [hunk 3]
```diff
@@ -1,40 +1,96 @@
...
 
-// CredentialContainer acts as an abstraction over both possible credential representations
+// Container acts as an abstraction over both possible credential representations
 // JWT representations are parsed upon container creation, while the original JWT is maintained
-type CredentialContainer struct {
+type Container struct {
 	// Credential ID
 	ID            string
 	Credential    *credential.VerifiableCredential
 	CredentialJWT *string
```
[the remark this answers is not in the document]
Comment (decentralgabe) [reply internal/credential/model.go:19]: added a type alias and utility methods

File: internal/credential/verification.go [hunk 4]
```diff
@@ -1,92 +1,144 @@
...
 	}
-	if err = v.verifier.VerifyCredential(credential); err != nil {
+
+	return v.staticVerificationChecks(credential)
+}
+
+// resolveIssuerKey resolves the issuer's public key from the credential's issuer DID.
+// TODO(gabe): perhaps this should be a verification method referenced on the proof object, not the issuer
+// TODO(gabe): support issuers that are not strings, but objects
+func (v Verifier) resolveIssuerKey(credential credsdk.VerifiableCredential) (kid string, pubKey crypto.PublicKey, err error) {
```
Comment (decentralgabe) [inline internal/credential/verification.go:103]: extracted this as well

**Pull request merged**